deploy-service/teletraanservice/src/main/java/com/pinterest/teletraan/config/ZKMysqlDataSourceFactory.java

```diff
@@ -84,9 +93,47 @@ public BasicDataSource build() throws Exception {
             host = reader.getHost(replicaSet);
             port = reader.getPort(replicaSet);
         }
-        KnoxDBKeyReader.init((role));
-        String userName = KnoxDBKeyReader.getUserName();
-        String password = KnoxDBKeyReader.getPassword();
-        return DatabaseUtil.createMysqlDataSource(host, port, userName, password, pool);
+
+        if (this.useMTLS) {
+            String userName = getUserNameFromSpiffeId();
+            String password = "password";
+            Map<String, String> proxyConnectionProps = ImmutableMap.<String, String>builder()
+                // ssl properties
+                .put("sslMode", "VERIFY_CA" )
+                .put("trustCertificateKeyStoreUrl", "file:/var/lib/normandie/fuse/jkstrust/generic" )
+                .put("trustCertificateKeyStoreType", "JKS" )
+                .put("trustCertificateKeyStorePassword", "pintastic" )
+                .put("clientCertificateKeyStoreUrl", "file:/var/lib/normandie/fuse/jks/generic" )
+                .put("clientCertificateKeyStoreType", "JKS" )
+                .put("clientCertificateKeyStorePassword", "pintastic" )
```

Comment:
The "pintastic" field is Pinterest specific, does this repo contain other Pinterest specific logic?

Reply:
yes, the password is supposed to be the same for any db connection in Pinterest.
I am not sure if we should hide them in the public repo. cc @ankilosaurus

Reply:
@liyaqin1 look at how we populate other db specific config. fr ex: knox key.

Reply:
I don't think we need to use knox key in this case, let me store these in script config if you think it is necessary ^^